core/src/main/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationService.scala

```diff
@@ -109,4 +112,44 @@ class GoogleProjectRegistrationService(protected val ctx: RawlsRequestContext,
         case None => Future.successful(None)
       }
     } yield finalResult
+
+  def getGoogleProjects(billingProjectName: Option[RawlsBillingProjectName]): Future[Seq[GoogleProjectRegistration]] = {
+    val accessibleGoogleProjectsFuture = samDAO
+      .listResourcesWithActions(SamResourceTypeNames.googleProject, SamGoogleProjectActions.readPolicies, ctx)
+      .map(_.map(resource => GoogleProjectId(resource.getResourceId)).toSet)
+
+    accessibleGoogleProjectsFuture.flatMap { accessibleGoogleProjects =>
+      googleProjectRegRepo.getGoogleProjectRegistrations(accessibleGoogleProjects).map { registrations =>
+        val filteredRegistrations = billingProjectName
+          .map(name => registrations.filter(_.billingProjectId == name))
+          .getOrElse(registrations)
```

Comment:
this queries the db to return all `google-project` rows the user can see, then immediately discards all google-projects which don't match the requested billing project. Instead, could you include the billing project in the db query where clause, so the db only returns the relevant rows?